File: GotravelDesktop/src/main/java/com/gotravel/Model/Viaje.java
package com.gotravel.Model;

import com.gotravel.Utils.Fechas;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Viaje {

    private int id;
    private String nombre;
    private String descripcion;
    private String fechaInicio;
    private String fechaFin;
    private Double costeTotal;
    private List<Etapa> etapas;

    public Viaje(String nombre, String descripcion, String fechaInicio, String fechaFin, Double costeTotal) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fechaInicio = fechaInicio;
        this.fechaFin = fechaFin;
        this.costeTotal = costeTotal;
    }

    public Viaje(int id, String nombre, String descripcion, String fechaInicio, String fechaFin, Double costeTotal) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fechaInicio = fechaInicio;
        this.fechaFin = fechaFin;
        this.costeTotal = costeTotal;
    }

    public String inicio() {
        return LocalDate.parse(this.fechaInicio, Fechas.formatoFromDb).format(Fechas.formatoFinal);
    }

    public String fin() {
        return LocalDate.parse(this.fechaFin, Fechas.formatoFromDb).format(Fechas.formatoFinal);
    }

}

File: GoTravelMobile/app/src/main/java/com/gotravel/mobile/data/model/Viaje.kt
package com.gotravel.mobile.data.model

import android.os.Build
import androidx.annotation.RequiresApi
import com.gotravel.mobile.ui.utils.formatoFinal
import com.gotravel.mobile.ui.utils.formatoFromDb
import kotlinx.serialization.Serializable
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDate

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Serializable
data class Viaje (
    val id: Int? = null,
    val nombre: String,
    val descripcion: String?,
    val fechaInicio: String,
    val fechaFin: String,
    val costeTotal: Double,
) {
    val inicio: String
        @RequiresApi(Build.VERSION_CODES.O)
        get() {
            return LocalDate.parse(fechaInicio, formatoFromDb).format(formatoFinal)
        }

    val final: String
        @RequiresApi(Build.VERSION_CODES.O)
        get() {
            return LocalDate.parse(fechaFin, formatoFromDb).format(formatoFinal)
        }
}